crates/rune/src/compile/v1/assemble.rs

```diff
@@ -1252,7 +1252,18 @@ fn expr_assign(
                         expr(field_access.expr, c, Needs::Value)?.apply(c)?;
                         c.scopes.decl_anon(span)?;
 
+                        #[cfg(feature = "dynamic_fields")]
+                        c.asm.push(
+                            if c.options.dynamic_fields {
+                                Inst::ObjectIndexDynamicSet { slot }
+                            } else {
+                                Inst::ObjectIndexSet { slot }
+                            },
+                            span,
+                        );
+                        #[cfg(not(feature = "dynamic_fields"))]
                         c.asm.push(Inst::ObjectIndexSet { slot }, span);
```

Comment:
Consider moving into a function which is feature gated with two different implementations instead?

Block-level feature gates are a bit awkward to deal with in my experience.

Something like:

```rust
push_field_lookup(c, slot);
```